scripts/ci/check-all-notebooks-are-tested.py

```diff
@@ -0,0 +1,42 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Check all notebooks are listed in `scripts/config/notebook-testing.toml` so
+they don't slip through our CI tests.
+"""
+
+from pathlib import Path
+import tomllib
+import sys
+
+config_path = Path("scripts/config/notebook-testing.toml")
+config = tomllib.loads(config_path.read_text())
+
+categorized_notebooks = set()
+for group in config.values():
+    for path in group:
+        categorized_notebooks.add(Path(path))
```

Comment:
Nit, this can be a comprehension

```python
categorized_notebooks = {
    Path(path)
    for group in config.values()
    for path in group
}
```